pageserver/src/waldecoder.rs

```diff
@@ -110,15 +111,16 @@ impl WalStreamDecoder {
 
                 self.lsn += SizeOfXLogLongPHD as u64;
                 continue;
-            } else if self.lsn % (XLOG_BLCKSZ as u64) == 0 {
+            } else if self.lsn.0 % (XLOG_BLCKSZ as u64) == 0 {
```

Comment:
I could use some help here.

What does `XLOG_BLCKSZ` mean?
What does `lsn % XLOG_BLCKSZ` mean?
Should this be a member function, e.g. `Lsn::block_offset(block_size: u64)`?
If so, is block size a compile-time constant? Is it always a power of 2?

Reply:
XLOG_BLCKSZ is mostly the same as BLCKSZ: size of page in Postgres. Postgres is mostly using bock IO for accessing files. lsn % XLOG_BLCKSZ just mean offset in the block. BLCKSZ and XLOG_BLCKSZ is compile time constants, chaging them requires recompilation of postgres and ... although it is possible, nobody is doing. SO them can be considered as  "world constants" as PI. Yes, them should be power of two.